modules/heat_conduction/src/materials/GapConductance.C

```diff
@@ -160,12 +160,6 @@ GapConductance::GapConductance(const InputParameters & parameters)
 }
 
 void
-GapConductance::initialSetup()
-{
-  setGapGeometryParameters(_pars, _coord_sys, _gap_geometry_type, _p1, _p2);
```

Comment:
My guess is that `_gap_geometry_type`, `_p1`, and `_p2` need to be declared as restartable data.  Since there are a few other things to fix, see if making them restartable works with the original code structure.